### PR TITLE
Add Hook To Modify Entity in StoreContext::injectStoredValue

### PR DESCRIPTION
**Pull request by TaysirTayyab**
# Overview
Due to some issues with Laravel and Model relationships, we need a way to reload a model that's stored in the key store. This PR creates the infrastructure to allow us to hook into the `injectStoredValue` method to manipulate the entity before accessing the property of it.

# Changes
```
bin/phpunit
bin/composer
```
- these are now version locked to a particular image of phpunit and composer

```
composer.json
composer.lock
```
- adding phpunit and all it's dependencies to the dev environment

```
StoreContext.php
StoreContextInterface.php
```
- adding a `callable` to the `injectStoredValues` method to serve as "hook" for the entity

```
phpunit.xml
StoreContextTest.php
```
- new unit tests for `StoreContext::injectStoredValues`

### CODE REVIEW

File: src/Behat/FlexibleMink/Context/StoreContext.php [hunk 1]
```diff
@@ -98,18 +98,33 @@ protected function getThingProperty($key, $property, $nth = null)
...
+            throw new Exception('Method $onGetFn must take one argument!');
+        }
+
         preg_match_all('/\(the ([^\)]+) of the ([^\)]+)\)/', $string, $matches);
         foreach ($matches[0] as $i => $match) {
             $thingName = $matches[2][$i];
             $thingProperty = str_replace(' ', '_', strtolower($matches[1][$i]));
 
-            if (!$thing = $this->get($thingName)) {
+            if (!$this->isStored($thingName)) {
```
Comment (awesomeousPrime) [inline src/Behat/FlexibleMink/Context/StoreContext.php]: We may be able to replace these lines with `assertIsStored`

- [ ] Done
- [ ] Approved

**Comment by medology-bugman** [comment]
Paging @johnnie502 to review the back-end work in this pull request!

I couldn't recognize who these files were relevant to:
bin/composer
bin/phpunit
composer.json
composer.lock
phpunit.xml

File: tests/Behat/FlexibleMink/Context/StoreContextTest.php [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+            $thing->test_property_1 = 'overwritten';
+
+            return $thing;
+        };
+        $this->assertEquals(
+            'overwritten',
+            $this->injectStoredValues('(the test_property_1 of the testObj)', $goodFn)
+        );
+    }
+}
```
Comment (awesomeousPrime) [inline tests/Behat/FlexibleMink/Context/StoreContextTest.php]: Cool!

**Comment by awesomeousPrime** [comment]
!reassign testing

**Comment by medology-bugman** [comment]
Paging @awesomeousPrime to review the testing work in this pull request!

**Comment by awesomeousPrime** [comment]
!reassign testing

**Comment by medology-bugman** [comment]
Paging @joshmedeski to review the testing work in this pull request!

**Comment by awesomeousPrime** [comment]
Testing review 👍  3113d95

**Comment by TaysirTayyab** [comment]
doing reassignments based on the people that reviewed.

!reassign back-end @awesomeousPrime 
!reassign testing @Chekote

**Comment by medology-bugman** [comment]
Paging @awesomeousPrime to review the back-end work in this pull request!

**Comment by TaysirTayyab** [comment]
!reassign testing @Chekote

**Comment by medology-bugman** [comment]
Paging @Chekote to review the testing work in this pull request!

**Comment by TaysirTayyab** [comment]
based on feedback from @Chekote and @awesomeousPrime, back-end and testing are approved.

File: src/Behat/FlexibleMink/Context/StoreContext.php (hunk 1 above)
Comment (TaysirTayyab) [inline src/Behat/FlexibleMink/Context/StoreContext.php]: `assertIsStored` throws an exception if it's not stored :(

**Pull request merged**